src/main/java/reactor/core/publisher/EventLoopProcessor.java

```diff
@@ -316,6 +313,33 @@ static long getAndSub(RingBuffer.Sequence sequence, long toSub) {
 	}
 
 	/**
+	 * A method to extract a name from the ThreadFactory if it turns out to be a Supplier
+	 * (in which case the supplied value string representation is used). Otherwise return
+	 * the current class's simpleName.
+	 *
+	 * @param threadFactory the factory to test for a supplied name
+	 * @param clazz
+	 * @return the name to use in thread pools
+	 */
+	protected static String defaultName(ThreadFactory threadFactory, Class<? extends EventLoopProcessor> clazz) {
+		String name = threadFactory instanceof Supplier ? ((Supplier)
+				threadFactory).get().toString() : null;
+		return null != name ? name : clazz.getSimpleName();
+	}
+
+	/**
+	 * A method to create a suitable default {@link ExecutorService} for use in implementors
+	 * {@link #requestTask(Subscription)} (a {@link Executors#newCachedThreadPool() cached
+	 * thread pool}), reusing a main name and appending {@code [request-task]} suffix.
+	 *
+	 * @param name the main thread name used by the processor.
+	 * @return a default {@link ExecutorService} for requestTask.
+	 */
+	protected static ExecutorService defaultRequestTaskExecutor(String name) {
+		return Executors.newCachedThreadPool(r -> new Thread(r,name+"[request-task]"));
```

Comment:
not if we want to provide the exact same underlying semantics of being able to start several threads per several requestTasks (executing in parallel).